app/server.py

```diff
@@ -59,14 +60,17 @@ def search(search_engine):
             error = [400, 'Not Found - missing query', qformat]
             return bad_request(error)
 
-        result = feed_gen(query, engine, count)
+        # first see if we can get the results for the cache
+        engine_and_query = engine + ':' + query
+        result = lookup(engine_and_query)
```

Comment:
This would cause issues with number of results

Reply:
I thought you already deferred that issue.  https://github.com/fossasia/query-server/pull/390#discussion_r157175500

Reply:
Cool then